src/Epic/index.stories.tsx

```diff
@@ -0,0 +1,101 @@
+    highlightedText: string;
+    buttonText: string;
+    buttonUrl: string;
+    paragraphs: Array<string>;
+    slotName?: string;
+    componentName?: string;
+};
+
+const TOTAL_PARAGRAPHS = 9;
+
+const content = {
+    heading: 'Since you’re here...',
+    paragraphs: [
+        '... we have a small favour to ask. More people, like you, are reading and supporting the Guardian’s independent, investigative journalism than ever before. And unlike many news organisations, we made the choice to keep our reporting open for all, regardless of where they live or what they can afford to pay.',
+        'The Guardian will engage with the most critical issues of our time – from the escalating climate catastrophe to widespread inequality to the influence of big tech on our lives. At a time when factual information is a necessity, we believe that each of us, around the world, deserves access to accurate reporting with integrity at its heart.',
+        'Our editorial independence means we set our own agenda and voice our own opinions. Guardian journalism is free from commercial and political bias and not influenced by billionaire owners or shareholders. This means we can give a voice to those less heard, explore where others turn away, and rigorously challenge those in power.',
+        'We hope you will consider supporting us today. We need your support to keep delivering quality journalism that’s open and independent. Every reader contribution, however big or small, is so valuable. ',
+    ],
+    highlightedText:
+        'Support The Guardian from as little as %%CURRENCY_SYMBOL%%1 - and it only takes a minute. Thank you.',
+    buttonText: 'Support The Guardian',
+    buttonUrl: 'https://support.theguardian.com/contribute',
+};
+
+const buildProps = (data: DataFromKnobs): EpicProps => {
+    return {
+        variant: {
+            heading: data.heading,
+            paragraphs: data.paragraphs,
+            highlightedText: data.highlightedText,
+            cta: {
+                text: data.buttonText,
+                baseUrl: data.buttonUrl,
+            },
+        },
+        tracking: {},
+    };
+};
+
+export const defaultStory = (): ReactElement => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const slotName = text('slotName', 'Epic');
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const componentName = text('componentName', 'Epic');
+    const heading = text('heading', content.heading);
+    const highlightedText = text('highlightedText', content.highlightedText);
+    const buttonText = text('buttonText', content.buttonText);
+    const buttonUrl = text('buttonUrl', content.buttonUrl);
+    const paragraphs = [];
+
+    // Add existing paragraphs as knobs, and then pad out with empty knobs
+    for (let i = 0; i < TOTAL_PARAGRAPHS; i++) {
+        const name = `paragraph${i + 1}`;
+        if (content.paragraphs[i]) {
+            paragraphs.push(text(name, content.paragraphs[i]));
```

Comment:
It took me a second to figure out what was going on here. So this works because `content.paragraphs[i]` is the default text provided further up. I wonder if it would be worth assigning `content.paragraphs[i]` to a const so we can give it a name, like `const defaultText = content.paragraphs[i]` just to make it super clear what's happening.

Reply:
I'll do some renaming and hopefully make things clearer!